File: 744.寻找比目标字母大的最小字母.py
#
# @lc app=leetcode.cn id=744 lang=python3
#
# [744] 寻找比目标字母大的最小字母
#

# @lc code=start
class Solution:
    def nextGreatestLetter(self, letters: List[str], target: str) -> str:
        left = -1
        right = len(letters)
        while left+1 < right:
            mid = (left+right)//2
            if ord(letters[mid]) < ord(target)+1:
                left = mid
            else:
                right = mid
        if right == len(letters):
            return letters[0]
        else:
            return letters[right]
# @lc code=end

File: 744.寻找比目标字母大的最小字母.cpp
/*
 * @lc app=leetcode.cn id=744 lang=cpp
 *
 * [744] 寻找比目标字母大的最小字母
 */

// @lc code=start
class Solution {
public:
    char nextGreatestLetter(vector<char>& letters, char target) {
        int left = -1, right = letters.size();
        while (left+1 < right) {
            int mid = left + (right - left) / 2;
            if ((int)letters[mid] < (int)target+1) {
                left = mid;
            }
            else {
                right = mid;
            }
        }
        return right == letters.size() ? letters[0] : letters[right];
    }
};
// @lc code=end

